src/Remapping/distributed_remapping.jl

```diff
@@ -144,65 +219,482 @@ function Remapper(target_hcoords, target_zcoords, space)
             target_hcoords_pid_bitmask(target_hcoords, horizontal_topology, pid)
     end
 
-    # Extract the coordinate we own (as a MPI process). This will flatten the matrix.
+    # Extract the coordinates we own (as an MPI process). This will flatten the matrix.
     local_target_hcoords = target_hcoords[local_target_hcoords_bitmask]
 
-    horz_mesh = horizontal_topology.mesh
-
-    interpolation_coeffs = map(local_target_hcoords) do hcoord
-        quad = Spaces.quadrature_style(space)
-        quad_points, _ = Quadratures.quadrature_points(FT, quad)
-        return interpolation_weights(horz_mesh, hcoord, quad_points)
-    end
+    # Compute interpolation matrices
+    helems =
+        Meshes.containing_element.(Ref(horizontal_mesh), local_target_hcoords)
+    ξs_combined =
+        Meshes.reference_coordinates.(
+            Ref(horizontal_mesh),
+            helems,
+            local_target_hcoords,
+        )
+    num_hdims = length(ξs_combined[begin])
+    # ξs is a Vector of SVector{1, Float64} or SVector{2, Float64}
+    # Here we split the two dimensions because we want to compute the two interpolation matrices.
+    ξs_split = Tuple([ξ[i] for ξ in ξs_combined] for i in 1:num_hdims)
+
+    # Compute the interpolation matrices
+    quad = Spaces.quadrature_style(space)
+    quad_points, _ = Quadratures.quadrature_points(FT, quad)
+    Nq = Quadratures.degrees_of_freedom(quad)
+
+    local_horiz_interpolation_weights = Tuple(
+        ArrayType(Quadratures.interpolation_matrix(ξs, quad_points)) for
+        ξs in ξs_split
+    )
```

Comment:
```suggestion
    local_horiz_interpolation_weights = map(ξs-> ArrayType(Quadratures.interpolation_matrix(ξs, quad_points)), ξs_split)
    )
```
Another advantage of the type-preserving pattern is that everything follows as a user expects: if a user passes in `ξs_split::Vector`, they'll get vector containers back. If it's a Tuple, they'll get a `Tuple` back (and, depending on the situation, can result in zero allocations).

Reply:
Here, we don't want to preserve type: we want a Tuple. We use NTuple{1} vs NTuple{2} to distinguish the number of horizontal dimensions in the kernels.

Reply:
That's fine, then let's go with `local_horiz_interpolation_weights = map(ξs-> ArrayType(Quadratures.interpolation_matrix(ξs, quad_points)), ξs_split))` (`ξs_split` is a Tuple).

Reply:
Ok.